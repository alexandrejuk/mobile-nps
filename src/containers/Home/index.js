import React from 'react'
import PropTypes from 'prop-types'
import Translate from '../../locales'

import {
  Title,
  Card,
} from '../../components/'

import CoffeeSvg from '../../assets/icons/coffee.svg'
import bookSvg from '../../assets/icons/book.svg'
import Banner from '../../assets/images/banners/banner-4.svg'
import LoggoutSvg from '../../assets/icons/loggout.svg'
import styles from './style.module.css'

const Home = ({
  goToPage,
}) => {
  return (
    <div className={styles.container}>
      <div className={styles.header}>
        <div className={styles.headerLoggout}>
          <img src={LoggoutSvg} alt="loggout" />
        </div>
        <div className={styles.headerBanner}>
          <img src={Banner} alt="banner information" />
        </div>
      </div>
      <div className={styles.borderEffect} />
      <div className={styles.content}>
        <Title
          type="bold"
          text={Translate('home.title')}
        />
        <div className={styles.contentMenu}>
          <Card
            action={() => goToPage('equipments')}
            image={CoffeeSvg}
            title={Translate('home.equipmentButton')}
          />
          <Card
            action={() => goToPage('practices')}
            image={bookSvg}
            title={Translate('home.goodPracticesButton')}
          />
        </div>
      </div>
    </div>
  )
}

Home.propTypes = {
  goToPage: PropTypes.func.isRequired,
}

export default Home
