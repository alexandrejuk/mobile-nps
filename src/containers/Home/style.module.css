@import '../../assets/css/colors.css';

.container:not(:first-child) {
  padding: 20px 16px;
}

.header {
  background-color: var(--color-california);
  padding: 20px 16px 40px 16px;
}

.header .headerLoggout {
  text-align: right;
}

.headerBanner {
  position: relative;
  top: -40px;
  box-sizing: border-box;
  min-width: 175px;
  max-width: 300px;
  height: 200px;
  text-align: center;
  margin: auto;
}

.headerBanner img {
  width: 100%;
  max-width: 300px;
}

.borderEffect {
  position: relative;
  border-radius: 30px 30px 0 0;
  height: 40px;
  width: 100%;
  background-color: var(--color-white);
  top: -37px;
}

.content {
  position: relative;
  top: -50px;
  padding: 0 16px;
}

.contentMenu {
  display: flex;
  padding-top: 20px;
}

.contentMenu > * {
  box-sizing: border-box;
  width: calc((100vw - 60px) / 2);
}

.contentMenu > :first-child{
  margin-right: 10px;
}

@media screen and (min-width: 320px) {
  .contentMenu {
    height: 128px;
  }
}
