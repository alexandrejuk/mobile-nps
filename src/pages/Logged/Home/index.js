import React from 'react'
import PropTypes from 'prop-types'
import { withRouter } from 'react-router'
import HomeContainer from '../../../containers/Home'

import GAInitialize from '../../../utils/ga'

const Home = ({
  history,
}) => {
  GAInitialize('/logged/home')

  const gotToPage = page => history.push(`/logged/${page}`)

  return (
    <HomeContainer goToPage={gotToPage} />
  )
}

Home.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
}

export default withRouter(Home)
