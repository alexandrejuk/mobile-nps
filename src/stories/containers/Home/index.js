import React from 'react'
import { action } from '@storybook/addon-actions'

import HomeContainer from '../../../containers/Home'

export default {
  title: 'Containers/Home',
}

export const Default = () => (
  <HomeContainer goToPage={action('go to page')} />
)
