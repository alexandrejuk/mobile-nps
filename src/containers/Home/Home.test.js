import React from 'react'
import { render } from '@testing-library/react'

import Home from './'

describe('Home container', () => {
  it('should render correctly', async () => {
    const callback = jest.fn()
    const { container } = render(
      <Home
        closeModal={callback}
        goToPage={callback}
        logout={callback}
        show={false}
        showModal={callback}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
