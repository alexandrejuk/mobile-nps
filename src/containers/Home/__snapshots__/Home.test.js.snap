// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home container should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <div
        class="headerLoggout"
      >
        <img
          alt="loggout"
          src="loggout.svg"
        />
      </div>
      <div
        class="headerBanner"
      >
        <img
          alt="banner information"
          src="banner-4.svg"
        />
      </div>
    </div>
    <div
      class="borderEffect"
    />
    <div
      class="content"
    >
      <h1
        class="primary medium left title bold"
      >
        Read more
      </h1>
      <div
        class="contentMenu"
      >
        <div
          class="card"
          role="button"
          tabindex="0"
        >
          <div
            class="imageContainer"
          >
            <img
              alt="item card"
              src="coffee.svg"
            />
          </div>
          <div
            class="title"
          >
            <h1
              class="primary medium center title bold"
            >
              Equipments
            </h1>
          </div>
        </div>
        <div
          class="card"
          role="button"
          tabindex="0"
        >
          <div
            class="imageContainer"
          >
            <img
              alt="item card"
              src="book.svg"
            />
          </div>
          <div
            class="title"
          >
            <h1
              class="primary medium center title bold"
            >
              Practices Guide
            </h1>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
